test/fixtures/source-map/typescript-sourcemapping_url_string.js

```diff
@@ -0,0 +1,4 @@
+"use strict";
+const content = '/*# sourceMappingURL=';
+throw new Error('an exception.');
+//# sourceMappingURL=typescript-sourcemapping_url_string.js.map
```

Comment:
With a deeper dig, I found that this works because of the trailing mark `$` in the RegExp: i.e. the search begins from the tail of the source content.

If a statement of a string literal that contains a magic comment is present after the actual source map magic comment, this still matches the string literal instead of the actual magic comment.

But I assume we should always take the last magic comment present in the source content, and the magic comment always comes after the real source content. Maybe we can simply add the `$` mark in the RegExp instead.

Reply:
Thanks for the feedback. I did go down the root at the beginning but found out that when both `sourceMappingURL` and `sourceURL` are specified.
Ex: https://github.com/nodejs/node/blob/85b46e1cab0f91eb87ff30f14bee0b456a63c83b/test/fixtures/source-map/tabs.js#L55-L56

I did however, go back and amend the RegExp to use the global flag which makes this simpler.